src/chromium/interfaces/launchBrowser.js

```diff
@@ -4,21 +4,26 @@ import { isCurrentTabValidUrlScheme } from "Shared/backgroundScripts/validTab.js
 /**
  * Launches the Firefox variant. If Firefox is not installed, launch the Firefox download page.
  *
- * @param {*} tab The tab object to launch the browser with.
+ * @param {string} url The url to launch in the browser.
  * @param {boolean} usePrivateBrowsing  True if firefox should be launched in private browsing mode, false for normal mode.
  * @returns {boolean} True if the browser was launched, false otherwise.
  */
-export async function launchBrowser(tab, usePrivateBrowsing = false) {
+export async function launchBrowser(url, usePrivateBrowsing = false) {
   if (!(await getIsFirefoxInstalled())) {
     browser.tabs.create({ url: "https://www.mozilla.org/firefox/" });
     return false;
   }
 
   if (isCurrentTabValidUrlScheme) {
+    const currentTab = await browser.tabs.query({
+      active: true,
+      currentWindow: true,
+    });
+    const currentTabId = currentTab[0].id;
     if (usePrivateBrowsing) {
-      browser.tabs.update(tab.id, { url: "firefox-private:" + tab.url });
+      browser.tabs.update(currentTabId, { url: "firefox-private:" + url });
```

Comment:
Question: Is there any reason for reusing the current tab instead of opening a new one?

If there is a special reason for it, prepend a comment so that it is obvious why the current tab is replaced instead of a new tab being opened.

Reply:
In the [spec](https://docs.google.com/document/d/1jTxG4kyDsgwAVHYkDvHvABmO85-wk9vppJe6ikNe76A/edit) 1.9.1.4, it says to use update so that the request comes from the extension and not the page in order to only ask for permission once.

Unless you mean create a new tab and call update on that new tab, in which case I'm not sure which would be preferred.

Reply:
From the description it is clear that the intent is to associate the navigation with the extension (opposed to a web page), and that author of the document believed that `tabs.update` is the way to do so.

To open tabs, one can consider `tabs.create` or `tabs.update`. The former opens a new tab, whereas the latter starts a navigation in the current tab. There are pros and cons to each, but since the spec is so prescriptive I won't suggest changing it at this point.